### PR TITLE
187 score samples

### PR DESCRIPTION
**Pull request by shania-m**
closes: #187

---

## What
 * `score_samples` added to `rubicon_ml.sklearn.pipeline` and logs an experiment with a `score_samples` metric. Score_Samples are used across samples rather than an aggregate score that would be measured in `score()`. 

>  Score_samples on the other hand is used to score individual scores across samples. This could be sum or the mean of all these scores are used to calculate score() for many estimators; such as the `FactorAnalysis` Estimator, `PCA` estimator,`BayesianGaussianMixture` estimator, and `HalvingGridSearchCV` estimator.

## How to Test
  * pytest -m should pass all tests

### CODE REVIEW

File: rubicon_ml/sklearn/pipeline.py [hunk 1]
```diff
@@ -131,6 +132,34 @@ def score(self, X, y=None, sample_weight=None):
...
+            If not None, this argument is passed as sample_weight keyword argument to the
+            score method of the final estimator.
+        """
+        if experiment is None:
+            experiment = self.project.log_experiment(**self.experiment_kwargs)
+        self.experiment = experiment
+        score_samples = super().score_samples(X)
+
+        if type(score_samples) is ndarray:
+            score_samples = super().score_samples(X).tolist()
```
Comment (ryanSoley) [inline rubicon_ml/sklearn/pipeline.py:155]: what's going on here? why are we calling `score_samples` twice in this case?

Comment (shania-m) [reply rubicon_ml/sklearn/pipeline.py:155]: good catch, that should just say  `score_samples = score_samples.tolist()` and that's because rubicon can't log ndarrays

File: rubicon_ml/sklearn/pipeline.py [hunk 2]
```diff
@@ -131,6 +131,35 @@ def score(self, X, y=None, sample_weight=None):
...
+        X : iterable
+            Data to predict on. Must fulfill input requirements of first step of the pipeline.
+        y : iterable, optional
+            Targets used for scoring. Must fulfill label requirements for all steps of the pipeline.
+        sample_weight : list, optional
+            If not None, this argument is passed as sample_weight keyword argument to the
+            score method of the final estimator.
+        """
+        if experiment is None:
+            experiment = self.project.log_experiment(**self.experiment_kwargs)
```
Comment (ryanSoley) [inline rubicon_ml/sklearn/pipeline.py:149]: I think we should follow the same logic as `score` and check `self.experiment` too
